custom/openclinica/management/commands/odm_to_app.py

```diff
@@ -74,61 +99,78 @@ def get_subject_form_source(name):
         xform = XFormBuilder(name)
-        xform.new_question('name', 'Person ID')  # Subject's unique ID. aka "Screening Number", "Subject Key"
-        xform.new_question('subject_study_id', 'Subject Study ID')  # Subject number for this study
-        xform.new_question('dob', 'Date of Birth', data_type='date')
-        xform.new_question('sex', 'Sex', data_type='int', choices={1: 'Male', 2: 'Female'})
-        xform.new_question('enrollment_date', 'Enrollment Date', data_type='date')
+        xform.new_question(CC_SUBJECT_KEY, 'Person ID')  # Unique ID. aka "Screening Number", "Subject Key"
+        xform.new_question(CC_STUDY_SUBJECT_ID, 'Subject Study ID')  # Subject number for this study
+        xform.new_question(CC_DOB, 'Date of Birth', data_type='date')
+        xform.new_question(CC_SEX, 'Sex', data_type='select1', choices={1: 'Male', 2: 'Female'})
+        xform.new_question(CC_ENROLLMENT_DATE, 'Enrollment Date', data_type='date')
         return xform.tostring(pretty_print=True)
 
     def new_subject_module(self, app):
+
+        def add_reg_form_to_module(module_):
+            reg_form = module_.new_form('Register Subject', None)
+            reg_form.unique_id = make_uuid()
+            reg_form.source = self.get_subject_form_source('Register Subject')
+            reg_form.actions.open_case = OpenCaseAction(
+                name_path='/data/name',
+                external_id=None,
+                condition=FormActionCondition(type='always')
+            )
+            reg_form.actions.update_case = UpdateCaseAction(
+                update={
+                    'subject_study_id': '/data/subject_study_id',
+                    'dob': '/data/dob',
+                    'sex': '/data/sex',
+                    'enrollment_date': '/data/enrollment_date',
+                },
+                condition=FormActionCondition(type='always')
+            )
+
+        def add_edit_form_to_module(module_):
+            edit_form = module_.new_form('Edit Subject', None)
+            edit_form.unique_id = make_uuid()
+            edit_form.source = self.get_subject_form_source('Edit Subject')
+            edit_form.requires = 'case'
+            edit_form.actions.case_preload = PreloadAction(
+                preload={
+                    '/data/name': 'name',
+                    '/data/subject_study_id': 'subject_study_id',
+                    '/data/dob': 'dob',
+                    '/data/sex': 'sex',
+                    '/data/enrollment_date': 'enrollment_date',
+                },
+                condition=FormActionCondition(type='always')
+            )
+            edit_form.actions.update_case = UpdateCaseAction(
+                update={
+                    'name': '/data/name',
+                    'subject_study_id': '/data/subject_study_id',
+                    'dob': '/data/dob',
+                    'sex': '/data/sex',
+                    'enrollment_date': '/data/enrollment_date',
+                },
```

Comment:
Rather than repeating the same values you could just swap the keys and values:
```
preload = {....}
update = {v: k for k, v in preload.items()}
```